azure/services/aso/aso.go

```diff
@@ -45,12 +52,15 @@ const (
 // and deletion of resources using ASO.
 type Service struct {
 	client.Client
+
+	clusterName string
 }
 
 // New creates a new ASO service.
-func New(ctrlClient client.Client) *Service {
+func New(ctrlClient client.Client, clusterName string) *Service {
 	return &Service{
-		Client: ctrlClient,
+		Client:      ctrlClient,
+		clusterName: clusterName,
```

Comment:
will this always be cluster name or would there be a use case for making it more generic (eg. `ownerName`) so a resource can be "owned" by an AzureMachine or AzureMachinePool for example (as opposed to the cluster itself)?